### PR TITLE
Updated mints to use pinia stores for state management

### PR DESCRIPTION
**Pull request by cjbeery24**

### CODE REVIEW

**Comment by callebtc** [comment]
Incredible PR! Thank you

File: src/js/notify.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  message,
+  type = "null",
+  position = "top",
+  caption = null,
+  color = null
+) {
+  // failure
+  Notify.create({
+    timeout: 5000,
+    type: "nuill",
```
Comment (user411) [inline src/js/notify.js:84]: typo?

File: src/components/ChooseMint.vue [hunk 2]
```diff
@@ -33,16 +33,13 @@
...
   props: {
-    proofs: Array,
-    activeProofs: Array,
-    mints: Array,
     tickerShort: String,
-    activeMintUrl: String,
-    activateMint: Function,
   },
   data: function () {
     return {
```
Comment (ebrakke) [inline src/components/ChooseMint.vue:45]: Need to move `chosenMint` to computed state, otherwise the invoices flow is broken

**Review by ebrakke — COMMENTED** [review]
Looks like something is not right with the invoice checker as well. I can generate an invoice and pay it, but it's never registering as "complete".

File: src/stores/mints.js [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        if (this.mints.length === 0) {
+          this.mints = [{ url: url, balance: 0 }];
+        } else if (this.mints.filter((m) => m.url === url).length === 0) {
+          // we don't have this mint yet
+          this.mints.push({ url: url, balance: 0 });
+        }
+
+        await this.activateMint(url, verbose);
+      } catch (error) {
+        throw error;
```
Comment (callebtc) [inline src/stores/mints.js:51]: Bug fix from #11:

```suggestion
        // activation failed, we remove the mint again from local storage
        this.mints = this.mints.filter((m) => m.url != url);
        throw error;
```

File: src/stores/mints.js [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        this.keys = previousKeys;
+        this.keysets = previousKeysets;
+
+        let err_msg = "Could not connect to mint.";
+        if (error.message.length) {
+          err_msg = err_msg + ` ${error.message}.`;
+        }
+        await notifyError(err_msg, "Mint activation");
+        // we don't handle this error yet so it commented out
+        // throw error;
```
Comment (callebtc) [inline src/stores/mints.js:107]: Change from #11:

```suggestion
        throw error;
```

File: src/js/notify.js (hunk 1 above)
Comment (cjbeery24) [reply src/js/notify.js]: Fixed

File: src/components/ChooseMint.vue (hunk 2 above)
Comment (cjbeery24) [reply src/components/ChooseMint.vue:45]: Fixed.

File: src/stores/mints.js (hunk 3 above)
Comment (cjbeery24) [reply src/stores/mints.js:53]: Merged it in.

File: src/stores/mints.js (hunk 4 above)
Comment (cjbeery24) [reply src/stores/mints.js]: Merged it in.

**Comment by callebtc** [comment]
LGTM

**Comment by callebtc** [comment]
Todo: 
- add `activateMint` to `useMintsStore` in `SettingsView.vue`
- move `this.splitToSend` in `melt` into `try, catch` block

**Comment by callebtc** [comment]
Invited you as a collaborator to this repo! 🙏

**Pull request merged**